### PR TITLE
Updating Microsoft.VisualStudio.TraceDataCollector source

### PR DESCRIPTION
**Pull request by fhnaseer**
Getting Microsoft.VisualStudio.TraceDataCollector from CodeCoverageExternals  instead of TestPlatformExternals,

### CODE REVIEW

File: scripts/build.ps1 [hunk 1]
```diff
@@ -569,6 +569,17 @@ function Publish-Package
 
     Copy-Item -Recurse $intellitraceSourceDirectory\* $intellitraceTargetDirectory -Force
     
+    # Copy Microsoft.VisualStudio.TraceDataCollector
+    $codeCoverageExternalsVersion = ([xml](Get-Content $env:TP_ROOT_DIR\scripts\build\TestPlatform.Dependencies.props)).Project.PropertyGroup.CodeCoverageExternalsVersion
+    $traceDataCollectorSourceDirectory = Join-Path $env:TP_PACKAGES_DIR "Microsoft.VisualStudio.TraceDataCollector\$codeCoverageExternalsVersion\lib"
+    $traceDataCollectorTargetDirectory = Join-Path $env:TP_OUT_DIR "$TPB_Configuration\Microsoft.VisualStudio.TraceDataCollector"
```
Comment (jakubch1) [inline scripts/build.ps1:575]: what is the purpose of this copy? Wondering why target contains folder Microsoft.VisualStudio.TraceDataCollector
we don't have such folder in vsix or nuget

File: src/package/nuspec/Microsoft.TestPlatform.nuspec [hunk 2]
```diff
@@ -258,15 +258,15 @@
...
     <file src="net451\$Runtime$\Extensions\Microsoft.VisualStudio.TestTools.DataCollection.MediaRecorder.Model.dll"	target="tools\net451\Common7\IDE\Extensions\TestPlatform\Extensions\Microsoft.VisualStudio.TestTools.DataCollection.MediaRecorder.Model.dll" />
     <file src="net451\$Runtime$\Extensions\Microsoft.VisualStudio.TestTools.DataCollection.VideoRecorderCollector.dll"	target="tools\net451\Common7\IDE\Extensions\TestPlatform\Extensions\Microsoft.VisualStudio.TestTools.DataCollection.VideoRecorderCollector.dll" />
-    <file src="net451\$Runtime$\Extensions\Microsoft.VisualStudio.TraceDataCollector.dll"	target="tools\net451\Common7\IDE\Extensions\TestPlatform\Extensions\Microsoft.VisualStudio.TraceDataCollector.dll" />
+    <file src="Microsoft.VisualStudio.TraceDataCollector\net472\Microsoft.VisualStudio.TraceDataCollector.dll"	target="tools\net451\Common7\IDE\Extensions\TestPlatform\Extensions\Microsoft.VisualStudio.TraceDataCollector.dll" />
```
Comment (cvpoienaru) [inline src/package/nuspec/Microsoft.TestPlatform.nuspec:269]: Should this be signed too ? (i.e. included in the `sign.proj` like [here](src/package/sign/sign.proj))

File: scripts/build.ps1 (hunk 1 above)
Comment (fhnaseer) [reply scripts/build.ps1:575]: This copies libs from nuget package to artifacts folder, and then nuspec file gets it from there,

File: src/package/nuspec/Microsoft.TestPlatform.nuspec (hunk 2 above)
Comment (jakubch1) [reply src/package/nuspec/Microsoft.TestPlatform.nuspec:269]: our binaries are already signed - so we can try avoiding it

File: scripts/build.ps1 (hunk 1 above)
Comment (jakubch1) [reply scripts/build.ps1:575]: did you check vsix if all tracedatacollector is correctly there?

File: src/package/nuspec/Microsoft.TestPlatform.nuspec (hunk 2 above)
Comment (fhnaseer) [reply src/package/nuspec/Microsoft.TestPlatform.nuspec:269]: In `sign.proj` i see `CoreLib.Net.dll` for signing as well. Should i remove these?,

File: scripts/build.ps1 (hunk 1 above)
Comment (fhnaseer) [reply scripts/build.ps1:575]: yes i have,

File: src/package/nuspec/Microsoft.TestPlatform.nuspec (hunk 2 above)
Comment (jakubch1) [reply src/package/nuspec/Microsoft.TestPlatform.nuspec:269]: You could try to remove and then run TestPlatform.CI.Real from your branch and see if VS will complain when you do insertion, I'm also OK with merging it and fixing it if needed

**Review by jakubch1 — APPROVED** [review]

**Pull request merged**